### PR TITLE
feature: support cross data source relative references

### LINKED ISSUE #40: Use data source as root during imports

### PR DESCRIPTION
**Pull request by soofstad**
## What does this pull request change?
- support _cross package relative references_
- added documentation on supported reference formats
- added "\_blueprintPath\_" as a keyword for values to resolve
- use the "black" profile for isort pre-commit hook to avoid conflicts with black
- set black line width to 119

REGRESSION: We can no longer use "_id" placeholders for entity id's. _id is not handled in any special way, and need to be "correct" on disk.

## Issues related to this change

closes #40

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,7 +57,34 @@ app
...
 
-#### General
+### Supported reference syntax
+The CLI tool will understand and resolve these kind of type reference schemas during import.
+All values with one of these keys; `("type", "attributeType", "extends", "_blueprintPath_")` will be evaluated for resolution.
+
+```bash
+# DMSS
+dmss://datasource/package/entity
+dmss://datasource/package/subfolder/entity
```
Comment (netr0m) [inline README.md:67]:
```suggestion
sys://datasource/package/entity
sys://datasource/package/subfolder/entity
```
Leave as `sys` until we have resolved the issue regarding renaming to `dmss://`

File: dm_cli/bin/dm [hunk 2]
```diff
@@ -18,8 +18,8 @@ from dm_cli.application import (
 )
 from dm_cli.dmss import dmss_api
 from dmss_api.exceptions import ApiException
-from dm_cli.import_package import (ApplicationException, import_package_tree,
-                                package_tree_from_zip)
+from dm_cli.import_package import (ApplicationException, import_package_tree)
+from package_tree_from_zip import package_tree_from_zip
```
Comment (KristianKjerstad) [inline dm_cli/bin/dm:22]:
```suggestion
from dm_cli.package_tree_from_zip import package_tree_from_zip
```

**Review by KristianKjerstad — COMMENTED** [review]
should the version in setup.py be updated to deploy a new python package?

I can see that the blueprints/entities in test_app_dir_struct still follow the old relative reference format (where / represents datasource/rootPackage) should this also be updated to follow the new format?

**Review by eoaksnes — APPROVED** [review]
Look good, but there are some other comments.

File: README.md (hunk 1 above)
Comment (soofstad) [reply README.md]: fixed

File: dm_cli/bin/dm [hunk 3]
```diff
@@ -18,8 +18,8 @@ from dm_cli.application import (
 )
 from dm_cli.dmss import dmss_api
 from dmss_api.exceptions import ApiException
-from dm_cli.import_package import (ApplicationException, import_package_tree,
-                                package_tree_from_zip)
+from dm_cli.import_package import (ApplicationException, import_package_tree)
```
[the remark this answers is not in the document]
Comment (soofstad) [reply dm_cli/bin/dm:21]: fixed

**Pull request merged**